api/controllers/ResumeController.ts

```diff
@@ -0,0 +1,37 @@
+import { JsonController, UseBefore, Post, UploadedFile, BadRequestError } from 'routing-controllers';
+import path = require('path');
+import StorageService from '../../services/StorageService';
+import { UserAuthentication } from '../middleware/UserAuthentication';
+import ResumeService from '../../services/ResumeService';
+import { UserModel } from '../../models/UserModel';
+import { AuthenticatedUser } from '../decorators/AuthenticatedUser';
+import { File, MediaType, UpdateResumeResponse } from '../../types';
+
+@UseBefore(UserAuthentication)
+@JsonController('/resume')
+export class ResumeController {
+  private resumeService: ResumeService;
+
+  private storageService: StorageService;
+
+  constructor(resumeService: ResumeService, storageService: StorageService) {
+    this.resumeService = resumeService;
+    this.storageService = storageService;
+  }
+
+  @Post()
+  async updateResume(@UploadedFile('file',
+    { options: StorageService.getFileOptions(MediaType.RESUME) }) file: File,
+    @AuthenticatedUser() user: UserModel): Promise<UpdateResumeResponse> {
+    if (path.extname(file.originalname) !== '.pdf') throw new BadRequestError('Filetype must be \'.pdf\'');
+
+    const oldResumes = await this.resumeService.getUserResumes(user);
+    await this.storageService.deleteAtUrls(oldResumes.map((resume) => resume.url));
+
```

Comment:
I think instead of having a `deleteAtUrls`, we can just do a findOne query on resumes and delete that single resume. The reason being is we'd never really have the use case for deleting all other resumes for a user - for a "single resume per user" case, we just delete the previous one and upload the new one, and for a "multiple resumes per user" case, we would either just not delete any resumes, or delete single resumes within a category (e.g. if a user uploads a DEVOPS resume, and then another DEVOPS resume, we would delete the single previous DEVOPS resume.), depending on the requirements.

So I think it makes sense to keep it simple and just have a "deleteAtUrl" function that we're calling here, and not have a "deleteAtUrls" one that will always just run once.